test/parallel/test-fs-chmod-mask.js

```diff
@@ -89,6 +90,32 @@ function test(mode, asString) {
     fs.lchmodSync(link, input);
     assert.strictEqual(fs.lstatSync(link).mode & 0o777, mode);
   }
+
+  if (fs.lchmod) {
+    const link = path.join(tmpdir.path, `lchmod-read-only-src-${suffix}`);
+    const file = path.join(tmpdir.path, `lchmod-read-only-dest-${suffix}`);
+    fs.writeFileSync(file, 'test', { encoding: 'utf-8', mode: 0o444 });
+    fs.symlinkSync(file, link);
+
+    fs.lchmod(link, input, common.mustCall((err) => {
+      assert.ifError(err);
+      assert.strictEqual(
+        fs.lstatSync(link).mode & kMaskOnlyRead, kMaskOnlyRead
```

Comment:
Could you see if
```suggestion
        fs.lstatSync(link).mode, mode
```
works? I think it should.

Reply:
I tried it but fails with
```
- 41471
+ 511
```

Reply:
I'm not sure
I checked and `41471 & 0o777 == 0o777` so I'm not sure it that's a good result. We need a review from someone who understands POSIX better than me.

Reply:
ping @nodejs/fs

Reply:
any updates ?

Reply:
Usually trafic picks up after the weekend

Reply:
oh my bad, didn't know ✌️

Reply:
```javascript
(41471).toString(8) // '120777'
```

The `777` means `rwx` permission across the board, of course, which happens for symlinks a lot on various systems.
The `0` means no setgid/setuid/sticky-bit.
And the `12` means it's a symbolic link.

(And if that doesn't help because I'm misunderstanding what is being sought, apologies in advance.)